contracts/multisigs/GnosisSafeSameAddressMultisig.sol

```diff
@@ -31,40 +31,63 @@ error WrongNumOwners(uint256 expected, uint256 provided);
 /// @param provided Provided owner address.
 error WrongOwner(address provided);
 
+/// @dev Multisig transaction resulted in a failure.
+/// @param provided Provided multisig address.
+error MultisigExecFailed(address provided);
+
 /// @title Gnosis Safe Same Address - Smart contract for Gnosis Safe verification of an already existent multisig address.
 /// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
 contract GnosisSafeSameAddressMultisig {
     // Default data size to be parsed as an address of a Gnosis Safe multisig proxy address
+    // This exact size suggests that all the changes to the multisig have been performed and only validation is needed
     uint256 public constant DEFAULT_DATA_LENGTH = 20;
 
-    /// @dev Verifies the existent gnosis safe multisig for changed owners and threshold.
-    /// @notice The multisig is supposedly updated before reaching this step such that new multisig is not created.
+    /// @dev Updated and verifies the existent gnosis safe multisig for changed owners and threshold.
+    /// @notice The multisig could be updated before reaching this step such that new multisig is not created.
+    ///         If the multisig is not updated, then the data modifying it has to be provided.
     ///         Note that the order of owners' addresses in the multisig is in reverse order, see comments below.
-    /// @param owners Set of multisig owners.
-    /// @param threshold Number of required confirmations for a multisig transaction.
-    /// @param data Packed data containing address of an existent gnosis safe multisig.
+    /// @param owners Set of updated multisig owners.
+    /// @param threshold Updated number for multisig transaction confirmations.
+    /// @param data Packed data containing address of an existent gnosis safe multisig and a payload to call the multisig with.
     /// @return multisig Address of a multisig (proxy).
     function create(
         address[] memory owners,
         uint256 threshold,
         bytes memory data
```

Comment:
The data contains the proxy multisig address that was deployed, and the payload to call the `execTransaction()` on that multisig to modify it.